projects/libpng-proto/png_fuzz_proto.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto2";
+// Very simple proto description of the PNG format,
+// described at https://en.wikipedia.org/wiki/Portable_Network_Graphics
+
+message IHDR {
+  required uint32 width = 1;
+  required uint32 height = 2;
+  required uint32 other1 = 3;
+  required uint32 other2 = 4;  // Only 1 byte used.
+}
+
+message PLTE {
+  required bytes data = 1;
+}
+
+message IDAT {
+  required bytes data = 1;
+}
+
+message OtherChunk {
+  required uint32 type = 1;
```

Comment:
looks like if type gets into known_chunks it will unnecessary mutate auxtype

oneof chunk {
  uint8 known_type = 1;
  uint32 auxtype = 2;
}

or 
message OtherChunk {
   required bool use_known_chunk_type = 1;
   required uint32 type = 2; // if (use_known_chunk_type) known_chunks[type()% arraysize(known_chunks)]
  ...

Reply:
done, using oneof. 
protos doesn't seem to support uint8 though.